### PR TITLE
Use xxhash for all BucketId hash sets and maps.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@vekterli Now xxhash details are hidden PTAL

File: searchcore/src/vespa/searchcore/bmcluster/bucket_db_snapshot.cpp [hunk 1]
```diff
@@ -66,9 +65,4 @@ BucketDbSnapshot::try_get_bucket_info(BucketId bucket_id) const
 
 }
 
-namespace vespalib {
-
-template class hash_map<BucketId, BucketInfo, BucketId::hash>;
-template class hash_set<BucketId, BucketId::hash>;
-
-}
+VESPALIB_HASH_MAP_INSTANTIATE_H(document::BucketId, storage::spi::BucketInfo, document::BucketId::hash);
```
Comment (vekterli) [inline searchcore/src/vespa/searchcore/bmcluster/bucket_db_snapshot.cpp:68]: Nit: missing newline

**Review by vekterli — APPROVED** [review]
:+1:

File: searchcore/src/vespa/searchcore/bmcluster/bucket_db_snapshot.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/bmcluster/bucket_db_snapshot.cpp]: Fixed

**Pull request merged**